src/actions/config.js

```diff
@@ -9,6 +9,13 @@ export const CONFIG_SUCCESS = "CONFIG_SUCCESS";
 export const CONFIG_FAILURE = "CONFIG_FAILURE";
 export const CONFIG_MERGE = "CONFIG_MERGE";
 
+const getConfigUrl = () => {
+  const validTypes = { 'text/yaml': 'yaml', 'application/x-yaml': 'yaml' };
+  const configLink = document.querySelector('link[rel="cms-config-url"]');
+  const isValidType = link => link && validTypes[link.type];
+  return isValidType(configLink) ? get(configLink, 'href') : 'config.yml';
```

Comment:
This returns an empty path for the default if the user supplies a link without an href. It really should return 'config.yml' on any invalid form of the link.

Reply:
**Proposed:**

```
const getConfigUrl = () => {
  const validTypes = { 'text/yaml': 'yaml', 'application/x-yaml': 'yaml' };
  const configLinkEl = document.querySelector('link[rel="cms-config-url"]');
  const isValidLink = configLinkEl && validTypes[configLinkEl.type] && get(configLinkEl, 'href');
  if (isValidLink) {
    console.log(`Netlify CMS using config file: "${configUrl}"`);
    return get(configLinkEl, 'href');
  }
  return 'config.yml';
}
```